core/src/test/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -57,15 +58,25 @@
 
     private static Node NODE = null;
 
-    private static void reset() {
+    private static void reset() throws Exception {
         assert NODE != null;
         stopNode();
         startNode();
     }
 
-    private static void startNode() {
+    private static void startNode() throws Exception {
         assert NODE == null;
         NODE = newNode();
+        // we must wait for all events to be processed. otherwise the node might have started, elected itself master but might not yet have removed the
+        // SERVICE_UNAVAILABLE/1/state not recovered / initialized block
+        assertBusy(new Runnable() {
```

Comment:
how about just replacing this with a cluster health request with wait for green and assertTimeout?